eliot/tests/test_message.py

```diff
@@ -246,3 +248,41 @@ def test_newWithSerializer(self):
         serializer = object()
         msg = Message.new(serializer, x=1)
         self.assertIs(msg._serializer, serializer)
+
+
+
+class WrittenMessageTests(TestCase):
+    """
+    Tests for L{WrittenMessage}.
+    """
+
+    def test_as_dict(self):
+        """
+        L{WrittenMessage.as_dict} returns the dictionary that will be serialized
+        to the log.
+        """
+        log_entry = pmap({
+            'timestamp': 1,
+            'task_uuid': 'unique',
+            'task_level': [1],
+            'foo': 'bar',
+        })
+        self.assertEqual(WrittenMessage.from_dict(log_entry).as_dict(), log_entry)
+
+
+    def test_from_dict(self):
+        """
+        L{WrittenMessage.from_dict} converts a dictionary that has been
+        deserialized from a log into a L{WrittenMessage} object.
+        """
+        log_entry = pmap({
+            'timestamp': 1,
+            'task_uuid': 'unique',
+            'task_level': [1],
+            'foo': 'bar',
+        })
+        parsed = WrittenMessage.from_dict(log_entry)
+        self.assertEqual(parsed.timestamp, 1)
```

Comment:
Jean-Paul has a (stronger than I do) dislike of multiple assertions, so in Flocker we'd be doing a single assert:

```
self.assertEqual([parsed.timestamp, parsed.task_uuid, ...], [1, 'unique', ...])
```

I don't care that much though given it's all the same assert, really.